### PR TITLE
how-to: update server recommendations for tidb deployment

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Update the server recommendations for tidb deployment.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/3119, https://github.com/pingcap/docs-cn/pull/3125
- Other reference link(s):

### CODE REVIEW

**Comment by ran-huang** [comment]
@yilongrong PTAL. Thanks!

File: hardware-and-software-requirements.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ As an open source distributed NewSQL database with high performance, TiDB can be
 
 ## Server recommendations
 
-You can deploy and run TiDB on the 64-bit generic hardware server platform in the Intel x86-64 architecture. The requirements and recommendations about server hardware configuration for development, test and production environments are as follows:
+You can deploy and run TiDB on the 64-bit generic hardware server platform in the Intel x86-64 architecture. The requirements and recommendations about server hardware configuration (not including the resources occupied by the operating system itself) for development, test, and production environments are as follows:
```
Comment (TomShawn) [inline hardware-and-software-requirements.md:31]:
```suggestion
You can deploy and run TiDB on the 64-bit generic hardware server platform in the Intel x86-64 architecture. The requirements and recommendations about server hardware configuration (ignoring the resources occupied by the operating system itself) for development, test, and production environments are as follows:
```

**Comment by yilongrong** [comment]
LGTM

**Pull request merged**